### PR TITLE
fix: decouple local recording from needing a valid jwt

### PR DESCRIPTION
**Pull request by quitrk**

### CODE REVIEW

File: react/features/recording/functions.ts [hunk 1]
```diff
@@ -253,12 +253,12 @@ export function getRecordButtonProps(state: IReduxState) {
     const localRecordingEnabled = !localRecording?.disable && supportsLocalRecording();
 
     const dropboxEnabled = isDropboxEnabled(state);
-    const recordingEnabled = recordingService?.enabled || localRecordingEnabled || dropboxEnabled;
+    const recordingEnabled = recordingService?.enabled || dropboxEnabled;
 
-    if (isModerator) {
+    if (localRecordingEnabled) {
```
Comment (saghul) [inline react/features/recording/functions.ts:258]: also check for the platform, or it might show empty on mobile.

Comment (quitrk) [reply react/features/recording/functions.ts:258]: that check is already done with `supportsLocalRecording`

Comment (saghul) [reply react/features/recording/functions.ts:258]: excellent!

**Review by saghul — APPROVED** [review]

**Comment by shawnchin** [comment]
@quitrk Many thanks for this fix 🙇🏽 

I've just responded to the JaaS ticket with some details on a few related issues, but I got back a "Your message was quarantined by a policy" response so no idea if it will reach anyone.

Here's what I sent:

---

We have discovered some inconsistencies/bugs related to recording. 

In the current state where JWT has `recording=false` for all users, and `localRecording.disable=false` set in configOverwrite, we have identified the following issues:

**1. The visibility of "highlight moments" UI is inconsistently**

The highlight option is not present in toolbar, but it is visible in the conference info at the top _only for non-moderators_.

Non-moderators see this:
Cursor_and__8x8__Re__Enabling_local_recording_-_shawn_qwil_io_-_Qwil_Messenger_Mail

Moderators see this:
Cursor_and__8x8__Re__Enabling_local_recording_-_shawn_qwil_io_-_Qwil_Messenger_Mail

_(p.s. as a side note, the sharp corners on the left of the meeting timer vs rounded corners on the right looks really odd. This happens when the subject is hidden.)_

**2. "Start Recording" is visible and clickable even when "Recording" is not visible in toolbar.**
Cursor_and__8x8__Re__Enabling_local_recording_-_shawn_qwil_io_-_Qwil_Messenger_Mail

**3. When clicked, "Start recording" defaults to cloud recording even if user has no permissions**
Cursor_and__8x8__Re__Enabling_local_recording_-_shawn_qwil_io_-_Qwil_Messenger_Mail

As expected, it fails if the user attempts to start recording.
Cursor_and__8x8__Re__Enabling_local_recording_-_shawn_qwil_io_-_Qwil_Messenger_Mail
If we toggle the local recording switch, it does indeed do local recording. And it can no longer be untoggled.

None of these are critical issues, but it is confusing to our users. We've had 2 reports so far.

What we expected/assumed:

1. If both cloud and local recording is not available, then users should not be able to access highlight feature from either toolbar or conference info bar.
    * If the highlight-moment is meant to be shown even when recording is disabled, it should be consistent for both moderator and non-moderator. 
    * There should also not be a "Start recording" if recording is not enabled or if it was removed from toolbar.
2. If only local recording is enabled, the recording modal should not default to cloud recording.

**Comment by saghul** [comment]
Agreed on both accounts!

**Comment by saghul** [comment]
@quitrk can you PTAL?

**Pull request merged**